### PR TITLE
Resolve Private Domain Model List API Inaccessibility Issue

### LINKED ISSUE #360: Private Domain Environment Model List API Unreachable, Resulting in Brief White Screen
Hello there,

I've encountered an issue within our private domain environment where the model list API is inaccessible. This has led to a situation where, after attempting to access the model list, the screen turns white for a short period. Here are the details:

**Environment:**
- Private domain setup
- Accessed via [insert method of access here, e.g., web browser, app, etc.]

**Expected Behavior:**
Normally, when accessing the model list, the relevant information should be displayed without any interruptions.

**Actual Behavior:**
Instead, what happens now is that upon trying to access the model list, the screen goes white for a brief duration, indicating that the data cannot be retrieved.

**Steps to Reproduce:**
1. Go to the model list section in the private domain environment.
2. Observe the screen turning white and the lack of data presentation.

**Attempted Fixes:**
- [List any steps you've taken in an attempt to solve this, like clearing cache, checking connectivity, reviewing access permissions, etc.]

**Additional Information:**
- [Any other relevant information, such as whether this issue occurs consistently or only under certain conditions.]

Looking forward to any guidance or solutions you may offer.

Best,
[Your Name]

### PR DESCRIPTION
**Pull request by runjinz**
## **User description**
This pull request addresses the issue experienced in private domain environments where accessing the model list API led to a brief white screen. The changes include the following enhancements:

- Ensured forced refresh of model list after configuration is read.
- Removed unnecessary model menu registrations and fetch operations.
- Replaced 'modelMenus' with 'config.modelNames' to ensure consistency and reliability.
- Cleaned up conditionals for better readability and simplicity.
- Removed obsolete and redundant model list fetching code as well as extraneous usage of debugger.
- Simplified configuration initialization to improve reliability, including the removal of the axios dependency.

These changes should resolve the problem as described in issue [#360](https://github.com/devchat-ai/devchat/issues/360), preventing the white screen upon model list access and providing a smoother experience in a private domain setup.

Closes devchat-ai/devchat#360

___

## **Type**
bug_fix, enhancement

___

## **Description**
- Removed axios dependency from `App.tsx` and simplified the initial state setup.
- Replaced `modelMenus` with `config.modelNames` in `InputMessage` component to ensure consistency with the configuration.
- Removed redundant model menu fetching logic from `ChatPanel.tsx`.
- Improved conditional checks in `Config.tsx` using optional chaining for better error handling.
- Introduced a new `Model` type in `ConfigStore.ts` and added a method to fetch language models, enhancing the store's structure and functionality.
- Cleaned up `InputStore.ts` by removing unused code related to model menu registration.

___

## **Changes walkthrough**
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>App.tsx</strong><dd><code>Simplify App Initialization and Configuration</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/views/App.tsx
<li>Removed axios dependency and simplified initial state for readiness.<br> <li> Removed complex API key handling and model fetching logic.<br> <li> Added a call to refresh the model list after setting configuration.<br>

</details>
    

  </td>
  <td><a href="https://github.com/devchat-ai/devchat-gui/pull/37/files#diff-0400c4bdb75ac23f8b5a3d5ae34e19ef4e3e10006019b1227f9dbd53cd49855e">+3/-31</a>&nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>index.tsx</strong><dd><code>Refactor InputMessage to Use Config Model Names</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/views/components/InputMessage/index.tsx
<li>Removed references to modelMenus and replaced with config.modelNames.<br> <li> Updated conditional rendering logic for model menu based on the new <br>config.modelNames.<br>

</details>
    

  </td>
  <td><a href="https://github.com/devchat-ai/devchat-gui/pull/37/files#diff-7daad21e4d166893092aac761f77eac42d083ebbdc54572fe7bdfb7a66f8dcf8">+3/-4</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>ChatPanel.tsx</strong><dd><code>Remove Unused Model Menu Fetching in ChatPanel</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/views/pages/ChatPanel.tsx
<li>Removed unused model menu fetching logic.<br> <li> Added braces to if statement for better readability.<br>

</details>
    

  </td>
  <td><a href="https://github.com/devchat-ai/devchat-gui/pull/37/files#diff-ea00ff4132e3551fa504fc1702071b888f6fc6f9a993882b00e0e814d609957f">+1/-2</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>Config.tsx</strong><dd><code>Improve Config Page Robustness and Clarity</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/views/pages/Config.tsx
<li>Fixed conditional checks to use optional chaining for robustness.<br> <li> Updated model array mapping to use the new Model type structure.<br>

</details>
    

  </td>
  <td><a href="https://github.com/devchat-ai/devchat-gui/pull/37/files#diff-34a216408ded56087e1ff91c7ab4bc14cb3dfed0c7d14fcf937da2a4eda32e4d">+7/-9</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>ConfigStore.ts</strong><dd><code>Refactor ConfigStore with New Model Type and Fetch Logic</code>&nbsp; </dd></summary>
<hr>

src/views/stores/ConfigStore.ts
<li>Introduced new Model type and updated modelsTemplate to use this type.<br> <li> Added logic to fetch language models using axios and set them in the <br>store.<br> <li> Refactored configuration setting logic to be more concise and robust.<br>

</details>
    

  </td>
  <td><a href="https://github.com/devchat-ai/devchat-gui/pull/37/files#diff-550f627d36fac0ff17defb905afede462ad2c1ee04ad357061574450d9b54092">+178/-127</a></td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>InputStore.ts</strong><dd><code>Clean Up Unused Code in InputStore</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/views/stores/InputStore.ts
<li>Removed unused regModelMenus function.<br> <li> Cleaned up imports and unused code.<br>

</details>
    

  </td>
  <td><a href="https://github.com/devchat-ai/devchat-gui/pull/37/files#diff-8cbb4c13ad2906d7d0bb69e8ad0ab949d9dad0b2d5634c3670132201152fa55c">+1/-38</a>&nbsp; &nbsp; </td>
</tr>                    
</table></td></tr></tr></tbody></table>

### CODE REVIEW

**Comment by runjinz** [comment]
Preparing PR description...

**Comment by runjinz** [comment]
**[PR Description](https://github.com/devchat-ai/devchat-gui/pull/37)** updated to latest commit (https://github.com/devchat-ai/devchat-gui/commit/b2268514fa325b743d917ac4ceae94b0b6406fff)

File: src/views/stores/ConfigStore.ts [hunk 1]
```diff
@@ -29,141 +58,163 @@ export const ConfigStore = types
...
+      updateSettle: (value: boolean) => {
+        self.settle = value;
+      },
+      getDefaultModel: () => {
+        return self.defaultModel;
+      },
+      getLanguage: () => {
+        return self.config?.language;
+      },
+      getUserKey: () => {
```
Comment (yangbobo2021) [inline src/views/stores/ConfigStore.ts:75]: getUserKey，getAPIBase，可能放一个判断逻辑中好一些？比如，devchat api base会有默认值，但key可能没有默认值，假如这个时候设置了openai的key与api base，会不会获取到两者信息不匹配？担心获取到了devchat api base但key是openai的key.

File: src/views/stores/ConfigStore.ts [hunk 2]
```diff
@@ -29,141 +58,163 @@ export const ConfigStore = types
...
 
-        if (!newConfig.models[item.name]) {
-          newConfig.models[item.name] = {
-            ...currentModel,
-          };
-        } else {
-          newConfig.models[item.name] = {
-            ...currentModel,
-            ...newConfig.models[item.name],
+        self.modelsTemplate.forEach((item) => {
```
Comment (yangbobo2021) [inline src/views/stores/ConfigStore.ts:115]: 咱们现在从网上直接获取模型列表，setConfig函数似乎只有读取配置文件后才会执行，此时配置文件中可能有模型列表，也可能没有模型列表。如果没有模型列表，理论上来说应该保持UI上模型列表为空，如果配置中有模型列表，那么这里也不需要与模型模版发生什么关系。

**Pull request merged**